app/helpers/tag_helper.rb

```diff
@@ -15,6 +16,6 @@ def qrcode_tag(url, size: 1.8)
   private
 
   def qualified_page_name
-    "#{controller_path.tr('/', '-')}-#{controller.action_name}"
+    "#{formatted_controller_path(controller_path)}-#{controller.action_name}"
```

Comment:
```suggestion
    "#{controller_path.tr('/_', '-')}-#{controller.action_name}"
```

変更前のコードと比べると一文字追加だけなのでこちらでいいように思いました。

また、こちら元々ほとんどのページで使われていたわけなので、CSSのデザインに影響が出ないか @machida さんに念の為確認をとった方がいいかなと思います。

Reply:
@komagata 
こちらmachidaさんに確認したところ、`_`を`-`に変換する処理は不要とのことでした。
理由としては`_`を`-`に変換する処理を加えることで影響範囲が出てくるためです。

今回自分は以下のように、action_uncompletedのように`_`が入るコントローラーの場合、`-`で繋ぐように処理を行っていました。(`action_comlpeted`を`action-comlpeted`に変換するようにしていました)

```rb
# app/views/talks/_tabs.html.slim
= link_to talks_action_uncompleted_index_path, class: "page-tabs__item-link #{current_link(/^talks-action-uncompleted-index/)}" do
```

しかし既存の`_`を含むコントローラーは(`regular_event`など)は以下のように`_`を`-`に変換する処理が行われていません。

https://github.com/fjordllc/bootcamp/blob/ef6baf579fbd9e1504cfb9e1911612be548926f7/app/views/events/_tabs.html.slim#L8

なので、今回は既存の処理に合わせてしまって問題ないとのことでしたので、`_`を`-`に変換する処理を削除します。

Reply:
@komagata 
2点質問させていただきたいです。🙌

> 変更前のコードと比べると一文字追加だけなのでこちらでいいように思いました。

ここで使っている`formatted_controller_path(controller_path)}`は

> ここにも同じ処理があるので共通化すると良さそうです。
https://github.com/fjordllc/bootcamp/pull/6393#discussion_r1240391006

こちらのコメントで頂いたアドバイスを参考に、

```rb
# app/helpers/body_class_helper.rb

module BodyClassHelper
  def qualified_controller_name
    controller.controller_path.tr('/', '-')
  end
.
.
```

```rb
# app/helpers/tag_helper.rb
.
.
def qualified_page_name
    "#{controller_path.tr('/', '-')}-#{controller.action_name}"
end
.
.
```

上の2箇所の共通化として以下のように切り出したメソッドを使用しています。

```rb
# app/helpers/path_conversion_helper.rb`

module PathConversionHelper
  def formatted_controller_path(path)
    path.tr('/', '-')
  end
end
```

ここで行った共通化はkoamgataさんから頂いた以下のアドバイスとのずれはないでしょうか？

> ここにも同じ処理があるので共通化すると良さそうです。
https://github.com/fjordllc/bootcamp/pull/6393#discussion_r1240391006

こちらが1点目の質問になります。🙏

****

またずれがない場合、

> 変更前のコードと比べると一文字追加だけなのでこちらでいいように思いました。
https://github.com/fjordllc/bootcamp/pull/6393#discussion_r1247505797

こちらで提案していただいたように`formatted_controller_path`を使用するのではなく

```rb
 def qualified_page_name
 "#{formatted_controller_path(controller_path)}-#{controller.action_name}"
.
.
```

以下のように実装する方がいいのではないかという点について、

```rb
 def qualified_page_name
 "#{controller_path.tr('/_', '-')}-#{controller.action_name}"
.
.
```

自分としては`formatted_controller_path`を使わないのであれば共通化する必要はないのではないかと考えました。

理由としては、`app/helpers/tag_helper.rb`,`app/helpers/body_class_helper.rb`の2つのファイルでのみ`formatted_controller_path`メソッドを使用しており、`app/helpers/tag_helper.rb`内では以下の1箇所のみでしか使われていません。

```rb
# app/helpers/tag_helper.rb
.
.
  def qualified_page_name
    "#{formatted_controller_path(controller_path)}-#{controller.action_name}"
  end
```
そのためこの箇所で使わないのであれば`app/helpers/body_class_helper.rb`のみで`formatted_controller_path`を使うことになるので共通化は不要なのではないかと考えました。

なので以下のどちらを行えば良いか判断がつかないため、アドバイスをいただきたいです。🙏

- 共通化をせずに既存処理のまま使用する。
- 共通化を行い以下の箇所でも`formatted_controller_path`を使用する

```rb
# app/helpers/tag_helper.rb
.
.
  def qualified_page_name
    "#{formatted_controller_path(controller_path)}-#{controller.action_name}"
  end
```

自分の解釈などに誤りがあれば教えていただきたいです。🙌💦

Reply:
@OdenTakashi 

## 1点目
 
> ここで行った共通化はkoamgataさんから頂いた以下のアドバイスとのずれはないでしょうか？

ずれがあります。

新しくメソッドを増やすのではなく、下記のようにする方がいいように思います。

```ruby
def qualified_page_name
    "#{qualified_controller_name}-#{controller.action_name}"
end
```

## 2点目

> またずれがない場合、

1点目にずれがあるのでそれが解決してから話しましょう。

Reply:
@komagata 
ご確認いただきありがとうございます。🙌

> 新しくメソッドを増やすのではなく、下記のようにする方がいいように思います。

なるほどです。
komagataさんのレビューを参考に以下の修正を行いました。
- `formatted_controller_path`メソッドを削除し、`qualified_controller_name`メソッドで共通化を行いました

また2点目の質問ですが、こちらは削除を行った`formatted_controller_path`メソッドに関する質問でしたのでスルーしていただいて問題ありません🙏